frontend/src/utils.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Given a potential url, verify that it is a valid url
+ */
+export const isValidURL = (url) => {
+  try {
+    const potential = new URL(url);
+    // Verify that its only http(s)
+    return ['https:', 'http:'].includes(potential.protocol);
+  } catch (err) {
+    return false;
+  }
+};
+
+/**
+ * Decide if a given url is external or not. Assumes, url is already valid.
+ * Definition of external is anything not matching the host name.
+ */
+export const isExternalURL = (url) => {
+  const newUrl = new URL(url);
+  let currentHost;
+  if (process.env.NODE_ENV === 'development') {
```

Comment:
Not exactly a fan of this, but couldn't think of another way to get this to work. Since we're using `create-react` we have to pre-pend with env vars with `REACT_APP_X` to get it to show up.

Reply:
Looks like `process.env.TTA_SMART_HUB_URI` isn't set in production environments (or at least isn't in Sandbox). I'm getting `TypeError: Failed to construct 'URL': Invalid URL` on line 44.

Could we instead (in any environment) pulll currentHost with:

`const currentHost = window.location`